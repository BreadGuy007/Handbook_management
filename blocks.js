"use strict";

/**
 * Derived functions
 */
var getNextSibling = siblingGetter( 'next' );
var getPreviousSibling = siblingGetter( 'previous' );
var getTagType = getConfig.bind( null, 'tagTypes' );
var getTypeKinds = getConfig.bind( null, 'typeKinds' );

/**
 * Globals
 */
var config = {
	tagTypes: {
		'IMG': 'image',
		'H1': 'heading',
		'H2': 'heading',
		'H3': 'heading',
		'H4': 'heading',
		'H5': 'heading',
		'H6': 'heading',
		'default': 'paragraph'
	},
	typeKinds: {
		'paragraph': [ 'text' ],
		'heading': [ 'heading', 'text' ],
		'image': [ 'image' ],
		'default': []
	}
};
var editor = queryFirst( '.editor' );
var switcher = queryFirst( '.block-switcher' );
var switcherButtons = query( '.block-switcher .type svg' );
var blockControls = queryFirst( '.block-controls' );
var inlineControls = queryFirst( '.inline-controls' );
var insertBlockButton = queryFirst( '.insert-block__button' );
var imageFullBleed = queryFirst( '.block-image__full-width' );
var imageAlignNone = queryFirst( '.block-image__no-align' );
var imageAlignLeft = queryFirst( '.block-image__align-left' );
var imageAlignRight = queryFirst( '.block-image__align-right' );
var selectedBlock = null;


/**
 * Initialization
 */
window.addEventListener( 'click', clearBlocks, false );
editor.addEventListener( 'input', attachBlockHandlers, false );
editor.addEventListener( 'input', clearBlocks, false );
insertBlockButton.addEventListener( 'click', openBlockMenu, false );
window.addEventListener( 'mouseup', onSelectText, false );

attachBlockHandlers();
attachControlActions();

/**
 * Core logic
 */
function attachBlockHandlers() {
	getBlocks().forEach( function( block ) {
		block.removeEventListener( 'click', selectBlock, false );
		block.addEventListener( 'click', selectBlock, false );
	} );
}

function getBlocks() {
	return Array.prototype.concat.apply( [],
			[ 'p', 'h2', 'img' ].map( query ) );
}

function selectBlock( event ) {
	clearBlocks();
	event.stopPropagation();
	event.target.className += ' is-selected';

	selectedBlock = event.target;
	showControls( selectedBlock );
}

function clearBlocks() {
	getBlocks().forEach( function( block ) {
		block.className = block.className.replace( 'is-selected', '' );
	} );
	var selectedBlock = null;

	hideControls();
	hideMenu();
}

function showControls( node ) {
	// toggle block-specific switcher
	switcherButtons.forEach( function( element ) {
		element.style.display = 'none';
	} );
	var blockType = getTagType( node.nodeName );
	var switcherQuery = '.type-icon-' + blockType;
	queryFirst( switcherQuery ).style.display = 'block';

	// reposition switcher
	var position = node.getBoundingClientRect();
	switcher.style.opacity = 1;
	switcher.style.top = ( position.top + 18 + window.scrollY ) + 'px';

	// show/hide block-specific block controls
	var kinds = getTypeKinds( blockType );
	var kindClasses = kinds.map( function( kind ) {
		return 'is-' + kind;
	} ).join( ' ' );
	blockControls.className = 'block-controls ' + kindClasses;
	blockControls.style.display = 'block';

	// reposition block-specific block controls
	blockControls.style.top = ( position.top - 36 + window.scrollY ) + 'px';
	blockControls.style.maxHeight = 'none';
}

function hideControls() {
	switcher.style.opacity = 0;
	blockControls.style.display = 'none';
}

// Show popup on text selection
function onSelectText( event ) {
	event.stopPropagation();
	var txt = "";

	if ( window.getSelection ) {
		txt = window.getSelection();
	} else if ( document.getSelection ) {
		txt = document.getSelection();
	} else if ( document.selection ) {
		txt = document.selection.createRange().text;
	}

	// Show formatting bar
	if ( txt != '' ) {
		inlineControls.style.display = 'block';
		var range = txt.getRangeAt(0);
		var pos = range.getBoundingClientRect();
		var selectCenter = pos.width / 2;
		var controlsCenter = inlineControls.offsetWidth / 2;
		inlineControls.style.left = ( pos.left + selectCenter - controlsCenter ) + 'px';
		inlineControls.style.top = ( pos.top - 48 + window.scrollY ) + 'px';
	} else {
		inlineControls.style.display = 'none';
	}
}

function attachControlActions() {
	Array.from( switcher.childNodes ).forEach( function( node ) {
		if ( 'svg' !== node.nodeName ) {
			return;
		}

		var classes = node.className.baseVal;

		if ( 'up' === classes ) {
			node.addEventListener( 'click', function() {
				event.stopPropagation();
				swapNodes( selectedBlock, getPreviousSibling( selectedBlock ) );
				attachBlockHandlers();
				reselect();
			}, false );
		} else if ( 'down' === classes ) {
			node.addEventListener( 'click', function() {
				event.stopPropagation();
				swapNodes( selectedBlock, getNextSibling( selectedBlock ) );
				attachBlockHandlers();
				reselect();
			}, false );
		}
	} );

	imageFullBleed.addEventListener( 'click', setImageFullBleed, false );
	imageAlignNone.addEventListener( 'click', setImageAlignNone, false );
	imageAlignLeft.addEventListener( 'click', setImageAlignLeft, false );
	imageAlignRight.addEventListener( 'click', setImageAlignRight, false );
}

function reselect() {
	queryFirst( '.is-selected' ).click();
}

function swapNodes( a, b ) {
	if ( ! ( a && b ) ) {
		return false;
	}

	var parent = a.parentNode;
	if ( ! parent ) {
		return false;
	}

	// insert node copies before removal
	parent.replaceChild( b.cloneNode( true ), a );
	parent.replaceChild( a.cloneNode( true ), b );

	return true;
}

/**
 * Utility functions
 */
function siblingGetter( direction ) {
	var sibling = direction + 'Sibling';

	return function getAdjacentSibling( node ) {
		if ( null === node ) {
			return null;
		}

		if ( null === node[ sibling ] ) {
			return null;
		}

		if ( '#text' === node[ sibling ].nodeName ) {
			return getAdjacentSibling( node[ sibling ] );
		}

		return node[ sibling ];
	}
}

function openBlockMenu( event ) {
	event.stopPropagation();
	var menu = document.getElementsByClassName( 'insert-block__menu' )[0];
	menu.style.display = 'block';
	menu.addEventListener( 'click', function( event ) {
		event.stopPropagation();
	}, false );
}

function hideMenu() {
	var menu = document.getElementsByClassName( 'insert-block__menu' )[0];
	menu.style.display = 'none';
}

function setImageFullBleed( event ) {
	event.stopPropagation();
	var image = query('img')[0];
	console.log(image);
	image.className = 'is-selected align-full-bleed';
}

function setImageAlignNone( event ) {
	event.stopPropagation();
	var image = query('img')[0];
	console.log(image);
	image.className = 'is-selected';
}

function setImageAlignLeft( event ) {
	event.stopPropagation();
	var image = query('img')[0];
	image.className = 'is-selected align-left';
}

function setImageAlignRight( event ) {
	event.stopPropagation();
	var image = query('img')[0];
	image.className = 'is-selected align-right';
}

function l( data ) {
	console.log.apply( console.log, arguments );
	return data;
}

function query( selector ) {
	return Array.from( document.querySelectorAll( selector ) );
}

function queryFirst( selector ) {
	return query( selector )[ 0 ];
}

function getConfig( configName, tagName ) {
	return config[ configName ][ tagName ] ||
		config[ configName ].default;
}
