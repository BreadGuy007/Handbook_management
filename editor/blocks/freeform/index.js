const { html } = wp.blocks.query;

wp.blocks.registerBlock( 'core/freeform', {
	title: wp.i18n.__( 'Freeform' ),

	icon: 'text',

	category: 'common',

	attributes: {
		html: html()
	},

	edit( { attributes } ) {
		return <div contentEditable>{ attributes.html }</div>;
	},

	save( { attributes } ) {
		return attributes.html;
	}
} );

wp.blocks.setUnknownTypeHandler( 'core/freeform' );
