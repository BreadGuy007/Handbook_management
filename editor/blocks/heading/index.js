const Editable = wp.blocks.Editable;
const { html, prop } = wp.blocks.query;

function Heading( { nodeName = 'H2', children } ) {
	// nodeName.toLowerCase() is used to map DOM nodeName values to proper tag.
	return wp.element.createElement( nodeName.toLowerCase(), null, children );
}

wp.blocks.registerBlock( 'core/heading', {
	title: 'Heading',
	icon: 'heading',
	category: 'common',

	attributes: {
		value: html( 'h1,h2,h3,h4,h5,h6' ),
		headingType: prop( 'h1,h2,h3,h4,h5,h6', 'nodeName' )
	},

	edit( attributes, onChange ) {
		const { headingType = 'H2', value } = attributes;

		return (
				<Editable
					nodeName={ headingType }
					value={ value }
					onChange={ ( nextValue ) => onChange( { value: nextValue } ) } />

		);
	},

	save( attributes ) {
		const { headingType = 'H2', value } = attributes;
		return <Heading nodeName={ headingType }>{ value }</Heading>;
	}
} );