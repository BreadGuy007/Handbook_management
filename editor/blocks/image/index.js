const { attr, html } = wp.blocks.query;
const Editable = wp.blocks.Editable;

wp.blocks.registerBlock( 'core/image', {
	title: wp.i18n.__( 'Image' ),

	icon: 'format-image',

	category: 'common',

	attributes: {
		url: attr( 'img', 'src' ),
		alt: attr( 'img', 'alt' ),
		caption: html( 'figcaption' )
	},

	edit( { attributes, isSelected, setAttributes } ) {
		const { url, alt, caption } = attributes;

		return (
			<figure>
				<img src={ url } alt={ alt } />
				{ caption || isSelected ? (
					<Editable
						tagName="figcaption"
						placeholder={ wp.i18n.__( 'Write captionâ€¦' ) }
						value={ caption }
						onChange={ ( value ) => setAttributes( { caption: value } ) } />
				) : null }
			</figure>
		);
	},

	save( { attributes } ) {
		const { url, alt, caption } = attributes;
		const img = <img src={ url } alt={ alt } />;

		if ( ! caption ) {
			return img;
		}

		return (
			<figure>
				{ img }
				<figcaption dangerouslySetInnerHTML={ { __html: caption } } />
			</figure>
		);
	}
} );
