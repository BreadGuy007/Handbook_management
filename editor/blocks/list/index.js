const Editable = wp.blocks.Editable;
const { html, prop } = wp.blocks.query;

wp.blocks.registerBlock( 'core/list', {
	title: wp.i18n.__( 'List' ),
	icon: 'editor-ul',
	category: 'common',

	attributes: {
		listType: prop( 'ol,ul', 'nodeName' ),
		items: wp.blocks.query.query(
			'li',
			{
				value: html()
			}
		)
	},

	controls: [
		{
			icon: 'editor-alignleft',
			title: wp.i18n.__( 'Align left' ),
			isActive: ( { align } ) => ! align || 'left' === align,
			onClick( attributes, setAttributes ) {
				setAttributes( { align: undefined } );
			}
		},
		{
			icon: 'editor-aligncenter',
			title: wp.i18n.__( 'Align center' ),
			isActive: ( { align } ) => 'center' === align,
			onClick( attributes, setAttributes ) {
				setAttributes( { align: 'center' } );
			}
		},
		{
			icon: 'editor-alignright',
			title: wp.i18n.__( 'Align right' ),
			isActive: ( { align } ) => 'right' === align,
			onClick( attributes, setAttributes ) {
				setAttributes( { align: 'right' } );
			}
		}
	],

	edit( { attributes, setAttributes } ) {
		const { listType = 'ol', items = [], align } = attributes;
		const content = items.map( item => {
			return `<li>${ item.value }</li>`;
		} ).join( '' );

		return (
			<Editable
				tagName={ listType }
				onChange={ ( value ) => setAttributes( { content: value } ) }
				style={ align ? { textAlign: align, listStylePosition: 'inside' } : null }
				value={ content } />
		);
	},

	save( { attributes } ) {
		const { listType = 'ol', items = [] } = attributes;
		const children = items.map( ( item, index ) => (
			<li key={ index } dangerouslySetInnerHTML={ { __html: item.value } } />
		) );
		return wp.element.createElement( listType.toLowerCase(), null, children );
	}
} );
