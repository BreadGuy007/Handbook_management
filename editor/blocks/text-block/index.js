const { html } = wp.blocks.query;
const Editable = wp.blocks.Editable;

wp.blocks.registerBlock( 'core/text', {
	title: 'Text',
	icon: 'text',

	attributes: {
		value: html()
	},

	edit( attributes, onChange ) {
		return (
			<Editable
				value={ attributes.value }
				onChange={ ( value ) => onChange( { value } ) }
			/>
		);
	},

	save( attributes ) {
		return <p>{ attributes.value }</p>;
	}
} );
