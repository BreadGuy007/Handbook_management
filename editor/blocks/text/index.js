const { html } = wp.blocks.query;
const Editable = wp.blocks.Editable;

wp.blocks.registerBlock( 'core/text', {
	title: wp.i18n.__( 'Text' ),

	icon: 'text',

	category: 'common',

	attributes: {
		value: html( 'p' )
	},

	edit( { attributes, onChange } ) {
		return (
			<Editable
				tagName="p"
				value={ attributes.value }
				onChange={ ( value ) => onChange( { value } ) }
			/>
		);
	},

	save( { attributes } ) {
		return <p dangerouslySetInnerHTML={ { __html: attributes.value } } />;
	}
} );
