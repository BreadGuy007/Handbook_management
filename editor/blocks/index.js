import './freeform';
import './text';
import './list';

