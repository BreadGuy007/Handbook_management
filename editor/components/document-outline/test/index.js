/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { DocumentOutline } from '../';

describe( 'DocumentOutline', () => {
	const paragraph = createBlock( 'core/paragraph' );
	const headingH1 = createBlock( 'core/heading', {
		content: 'Heading 1',
		nodeName: 'H1',
	} );
	const headingParent = createBlock( 'core/heading', {
		content: 'Heading parent',
		nodeName: 'H2',
	} );
	const headingChild = createBlock( 'core/heading', {
		content: 'Heading child',
		nodeName: 'H3',
	} );

	describe( 'no header blocks present', () => {
		it( 'should not render when no blocks provided', () => {
			const wrapper = shallow( <DocumentOutline /> );

			expect( wrapper.html() ).toBe( null );
		} );

		it( 'should not render when no heading blocks provided', () => {
			const blocks = [ paragraph ];
			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );

			expect( wrapper.html() ).toBe( null );
		} );
	} );

	describe( 'header blocks present', () => {
		it( 'should match snapshot', () => {
			const blocks = [ headingParent, headingChild ];
			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );

			expect( wrapper ).toMatchSnapshot();
		} );

		it( 'should render an item when only one heading provided', () => {
			const blocks = [ headingParent ];
			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );

			expect( wrapper.find( 'TableOfContentsItem' ) ).toHaveLength( 1 );
		} );

		it( 'should render two items when two headings and some paragraphs provided', () => {
			const blocks = [ paragraph, headingParent, paragraph, headingChild, paragraph ];
			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );

			expect( wrapper.find( 'TableOfContentsItem' ) ).toHaveLength( 2 );
		} );

		it( 'should render warnings for multiple h1 heaedings', () => {
			const blocks = [ headingH1, paragraph, headingH1, paragraph ];
			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );

			expect( wrapper ).toMatchSnapshot();
		} );
	} );
} );
