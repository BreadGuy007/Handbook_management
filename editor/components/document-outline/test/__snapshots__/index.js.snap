// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentOutline header blocks present should match snapshot 1`] = `
<div
  className="document-outline"
>
  <ul>
    <TableOfContentsItem
      isValid={2}
      key="0"
      level={2}
      onClick={[Function]}
    >
      Heading parent
    </TableOfContentsItem>
    <TableOfContentsItem
      isValid={3}
      key="1"
      level={3}
      onClick={[Function]}
    >
      Heading child
    </TableOfContentsItem>
  </ul>
</div>
`;
